Asking for Help Lesson: Added Supplemental Links (#29571)
TheOdinProject/curriculum: foundations/introduction/asking_for_help.md

This section contains helpful links to related content. It isn't required, so consider it supplemental.

> - [Asking for help is hard — but people want to help more than we realize](https://news.stanford.edu/stories/2022/09/asking-help-hard-people-want-help-realize) — Stanford article on why asking for help feels difficult, and how others are often more willing to help than we assume.

Reviewer: This is a little repetitive, so I think it would be better to do something like this:

```suggestion
- An encouraging article from Stanford on [Asking for help is hard — but people want to help more than we realize](https://news.stanford.edu/stories/2022/09/asking-help-hard-people-want-help-realize).
```

Author: Sounds good!

> - [The Mental Battle behind Asking for Help](https://hbr.org/2023/07/why-its-so-hard-to-ask-for-help) — Harvard Business Review article exploring the psychological barriers to asking for assistance.

Reviewer: The wording needs to be updated here. I don't like the format that you are using here. Can you update this a complete sentence, something like this:
```suggestion
- Harvard Business Review article exploring [The Mental Battle behind Asking for Help](https://hbr.org/2023/07/why-its-so-hard-to-ask-for-help).
